File: 3d_printer.rb
X, Y, Z = gets.chomp.split.map(&:to_i)
S = []

Z.times do |z|
  S.push []
  X.times do
    S[z].push gets.chomp
  end
  gets
end

z = Z
while z > 0
  z -= 1
  t = ""
  Y.times do |y|
    t += X.times.any? { |x| S[z][x][y] == '#' } ? '#' : '.'
  end
  puts t
end

File: 3d_printer.py
X, Y, Z = map(int, input().split())
S = []
for z in range(Z):
    S.append([])
    for x in range(X):
        S[z].append(input())
    input()
while Z > 0:
    Z -= 1
    t = ""
    for y in range(Y):
        t += '#' if any(S[Z][x][y] == '#' for x in range(X)) else '.'
    print(t)